tests/worldmap/QuadTreeTest.cpp

```diff
@@ -0,0 +1,233 @@
+#include "../../src/worldmap/QuadTree.h"
+
+#include <iostream>
+
+#include <catch2/catch.hpp>
+
+namespace
+{
+double randNum(double low, double high)
+{
+	return low + (rand() / (RAND_MAX / (high - low))); // NOLINT(cert-msc50-cpp)
+}
+
+point_t randPoint(double size)
+{
+	return {randNum(-size / 2, size / 2), randNum(-size / 2, size / 2), 1};
+}
+} // namespace
+
+TEST_CASE("QuadTree - AddPoints", "[QuadTree]")
+{
+	srand(time(nullptr)); // NOLINT(cert-msc51-cpp)
+	QuadTree tree(100);
+
+	REQUIRE(tree.empty());
+
+	for (int i = 0; i < 50; i++)
+	{
+		const point_t &point = randPoint(100);
+		REQUIRE(tree.getSize() == i);
+		REQUIRE(tree.add(point));
+		REQUIRE(tree.getSize() == i + 1);
+	}
+
+	REQUIRE_FALSE(tree.empty());
+}
+
+TEST_CASE("QuadTree - RemovePoints", "[QuadTree]")
+{
+	srand(time(nullptr)); // NOLINT(cert-msc51-cpp)
+	QuadTree tree(100);
+
+	points_t points;
+
+	for (int i = 0; i < 50; i++)
+	{
+		const point_t &point = randPoint(100);
+		points.push_back(point);
+		REQUIRE(tree.getSize() == i);
+		REQUIRE(tree.add(point));
+		REQUIRE(tree.getSize() == i + 1);
+	}
+
+	int size = tree.getSize();
+	for (const point_t &point : points)
+	{
+		REQUIRE(tree.getSize() == size);
+		REQUIRE(tree.remove(point));
+		REQUIRE(tree.getSize() == --size);
+	}
+	REQUIRE(tree.getSize() == 0);
+	REQUIRE(tree.empty());
+}
+
+TEST_CASE("QuadTree - RemoveNonexistentPoints", "[QuadTree]")
+{
+	srand(time(nullptr)); // NOLINT(cert-msc51-cpp)
+	QuadTree tree(100, 4);
+
+	for (int i = 0; i < 50; i++)
+	{
+		// make the size smaller, so we can guarantee that some points are not in the tree
+		tree.add(randPoint(5));
+	}
+
+	auto size = tree.getSize();
+	REQUIRE_FALSE(tree.remove({53, 1, 1}));
+	REQUIRE_FALSE(tree.remove({0, 0, 0}));
```

Comment:
I'm not suuuper concerned about this, but is there a chance that this point might actually be in the tree?

Reply:
https://github.com/huskyroboticsteam/PY2020/blob/d0c4c596cfb738e6b03d628f134dd6e8b1c587af/tests/worldmap/QuadTreeTest.cpp#L70-L74

All the points added fall in the bounding box centered at (0,0,1) with a height and width of 5, so all of these points are guaranteed to fall outside of this range. (and {0,0,0} has an invalid z-coord, so that one doesn't work either)

Reply:
Oh right! I forgot that 0 is an invalid z coordinate.